scripts/db_access/client_side_db.py

```diff
@@ -227,18 +244,41 @@ def _main(soledad, km, args):
             yield _export_key(args, km, args.expoert_public_key, private=False)
         if args.export_incoming_messages:
             yield _export_incoming_messages(soledad, args.export_incoming_messages)
-    except:
-        pass
+    except Exception as e:
+        logger.error(e)
     finally:
-        reactor.stop()
+        soledad.close()
+        reactor.callWhenRunning(reactor.stop)
```

Comment:
The main function has the `inlineCallbacks` decorator, so the `reactor.stop` callback will only be added after all main function yields have returned. Anyway, I don't see what would be a difference between the way it is proposed (using `callWhenRunning`) and using `callLater` with a delay of 1 second.